### PR TITLE
Require or disallow a space before XYZ parentheses

### PR DESCRIPTION
**Pull request by keegan-lillo**
Implements #140

I wanted to throw this up here for review before I continued with the other at-rules that take arguments.

### CODE REVIEW

**Comment by kristerkari** [comment]
Thanks! I'll try to review this as soon as possible.

File: src/rules/at-function-parentheses-space-before/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  return (root, result) => {
+    const validOptions = utils.validateOptions(result, ruleName, {
+      actual: value,
+      possible: ["always", "never"]
+    });
+    if (!validOptions) {
+      return;
+    }
+
+    root.walkAtRules(decl => {
```
Comment (kristerkari) [inline src/rules/at-function-parentheses-space-before/index.js]: I think that you should be able to do the `function` filtering already here, e.g.
https://github.com/kristerkari/stylelint-scss/blob/34668406e422148029bada5d32ef6405d11b7c5f/src/rules/at-extend-no-missing-placeholder/index.js#L18

File: src/rules/at-function-parentheses-space-before/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      description: "No space before parentheses."
+    },
+    {
+      code: `
+      @function  foo($n) {
+      }
+    `,
+      line: 2,
+      message: messages.expected,
+      description: "Too many spaces before parentheses."
```
Comment (kristerkari) [inline src/rules/at-function-parentheses-space-before/__tests__/index.js]: This should probably be `"Extra spaces after @function."` ?

**Comment by kristerkari** [comment]
@keegan-lillo This looks really good, thanks a lot! I think that you can go ahead and implement the other rules if you plan to do that 👍

**Comment by kristerkari** [comment]
@keegan-lillo there were some issues with `.travis.yml` settings, which caused the Travis builds to fail. Could you rebase this branch against master?

**Comment by keegan-lillo** [comment]
@kristerkari This is ready to be looked at now :)

**Comment by kristerkari** [comment]
@keegan-lillo Great! I'll try to get this reviewed later today.

**Comment by kristerkari** [comment]
Looks really good! Could you also add `at-mixin-parentheses-space-before` and  `at-else-if-parentheses-space-before` to README?

I think that after that this can be merged.

**Comment by keegan-lillo** [comment]
@kristerkari Whoops! Fixed now along with some other grammar fixes

File: src/rules/at-else-if-parentheses-space-before/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# at-mixin-parentheses-space-before
```
Comment (kristerkari) [inline src/rules/at-else-if-parentheses-space-before/README.md]: I noticed that this should be `at-else-if-parentheses-space-before`, but I can merge this PR and fix it with a separate commit.

**Pull request merged**